registry/registry.go

```diff
@@ -38,24 +40,24 @@ type Registry struct {
  * This passes http.DefaultTransport to WrapTransport when creating the
  * http.Client.
  */
-func New(registryUrl, username, password string) (*Registry, error) {
+func New(ctx context.Context, registryUrl, username, password string) (*Registry, error) {
```

Comment:
Can you clarify how terraform uses the plugin? We'll need to put base image spec in terraform but I think hardcoding is sufficient. Do you know if that's the case?